enterprise/server/remote_execution/runner/runner.go

```diff
@@ -1384,7 +1385,10 @@ func (p *pool) TryRecycle(ctx context.Context, r interfaces.Runner, finishedClea
 	if !cr.PlatformProperties.RecycleRunner {
 		return
 	}
-	if !finishedCleanly || cr.doNotReuse {
+	p.mu.Lock()
+	state := cr.state
+	p.mu.Unlock()
+	if !finishedCleanly || cr.doNotReuse || state != ready {
```

Comment:
Not specific to this PR, just an area for improvement. Looking back at the [logs](https://console.cloud.google.com/logs/query;query=%22%2Fuploads%2Fd7f5b86c-1655-4ff7-88c9-2d77bcba92ea%2Fblobs%2Fblake3%2F3ca0fa8f254690f77ec5aef026c5f534e03ee16e70b41d117c457553bb079037%2F650%22;cursorTimestamp=2025-08-08T21:24:22.928695762Z;startTime=2025-08-08T17:30:16.458Z;endTime=2025-08-09T07:04:14.364Z?project=flame-build&inv=1&invt=Ab5YOA), we didn't log when `DownloadInputs` failed. I think we got that hint from the execution result via the UI. It may be confusing to see this log message mentioning the "previous execution error". Doesn't have to be this PR but maybe we should add additional logging about execution errors

Reply:
Ah, the panic is happening in a goroutine in `monitorVMContext`, so it would only affect that goroutine. Still not sure why we didn't log an error though :thinking: I can't see how the task could succeed, yet still try to call `Pause()` before calling `Create()`.

Reply:
nvm, the panic kills the executor ([thread](https://buildbuddy-corp.slack.com/archives/C06N9AHA4JX/p1755102191290779?thread_ts=1755026363.300329&cid=C06N9AHA4JX)). So I think what's happening is that there is a race - either `monitorVMContext` starts a goroutine that might panic and kill the executor before we have a chance to log the error.